src/components/svg_editor.tsx

```diff
@@ -0,0 +1,261 @@
+import type { Molecule } from 'openchemlib';
+import type {
+  FormEvent,
+  KeyboardEvent as ReactKeyboardEvent,
+  MouseEvent,
+} from 'react';
+import { useEffect, useMemo, useReducer, useRef, useState } from 'react';
+
+import type { SvgRendererProps } from './svg_renderer.js';
+import { SvgRenderer } from './svg_renderer.js';
+
+export interface SvgEditorProps
+  extends Omit<SvgRendererProps, 'atomHighlight'> {
+  onChange: (newMolecule: Molecule) => void;
+}
+
+type State =
+  | { mode: 'view' }
+  | {
+      mode: 'atom-label-edit';
+      atomId: number;
+      formCoords: { x: number; y: number };
+    };
+const initialState: State = { mode: 'view' };
+
+type Action =
+  | {
+      type: 'startEdit';
+      atomId: number;
+      event: MouseEvent;
+    }
+  | { type: 'stopEdit' };
+
+function stateReducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'startEdit': {
+      // Ignore if we are already in edit mode
+      if (state.mode !== 'view') return state;
+
+      const { clientX, clientY } = action.event;
+      const target = action.event.target as SVGCircleElement;
+      const svg = target.closest('svg') as SVGElement;
+      const rect = svg.getBoundingClientRect();
+      const formCoords = {
+        // offset by 5px to avoid cursor onMouseLeave not triggered
+        x: clientX - rect.x + 5,
+        y: clientY - rect.y + 5,
+      };
+      return { mode: 'atom-label-edit', atomId: action.atomId, formCoords };
+    }
+    case 'stopEdit':
+      return { mode: 'view' };
+    default:
+      // @ts-expect-error action type narrowing
+      throw new Error(`Unknown action type: ${action.type}`);
+  }
+}
+
+/**
+ * A component that renders an SVG editor for a given molecule.
+ * Under the hood it uses the SvgRenderer component to display the molecule and catch interactions.
+ * @param props - The props for the SvgEditor component.
+ * @returns JSX.Element
+ */
+export function SvgEditor(props: SvgEditorProps) {
+  const { molecule, onChange, ...svgProps } = props;
+  const [state, dispatch] = useReducer(stateReducer, initialState);
+  const [atomHighlight, setAtomHighlight] = useState<number>(-1);
+  const atomsHighlight = useMemo(() => {
+    if (atomHighlight === -1) return undefined;
+
+    return [atomHighlight];
+  }, [atomHighlight]);
+
+  const atomRef = useRef(atomHighlight);
+  const onChangeRef = useRef(onChange);
+  useEffect(() => {
+    atomRef.current = atomHighlight;
+    onChangeRef.current = onChange;
+  });
+  useEffect(() => {
+    if (state.mode !== 'view') return;
+
+    function onKeyDown(event: KeyboardEvent) {
+      if (event.key !== 'Backspace') return;
+      if (atomRef.current === -1) return;
+
+      const atomId = atomRef.current;
+      const newMolecule = molecule.getCompactCopy();
+      newMolecule.setAtomCustomLabel(atomId, null as never);
```

Comment:
If `null` is supported, this is a bug in the OCL types. Please open an issue and/or fix it.

Reply:
oh and we usually don't silence errors like that. We use `@ts-expect-error` so that if the error disappears in the future we can fix the code.